Algorithm.CSharp/EmaCrossAlphaModelFrameworkAlgorithm.cs

```diff
@@ -13,106 +13,63 @@
 {
     /// <summary>
     /// Framework algorithm that uses the <see cref="EmaCrossAlphaModel"/>.
     /// </summary>
-    public class EmaCrossAlphaModelFrameworkAlgorithm : QCAlgorithm, IRegressionAlgorithmDefinition
+    public class EmaCrossAlphaModelFrameworkAlgorithm : BaseAlphaModelFrameworkRegressionAlgorithm
     {
         public override void Initialize()
         {
-            SetStartDate(2013, 10, 07);
-            SetEndDate(2013, 10, 11);
-
-            var symbols = new[] { "SPY", "AIG", "BAC", "IBM" }
-                .Select(ticker => QuantConnect.Symbol.Create(ticker, SecurityType.Equity, Market.USA))
-                .ToList();
-
-            // Manually add SPY and AIG when the algorithm starts
-            SetUniverseSelection(new ManualUniverseSelectionModel(symbols.Take(2)));
-
-            // At midnight, add all securities every day except on the last data
-            // With this procedure, the Alpha Model will experience multiple universe changes
-            AddUniverseSelection(new ScheduledUniverseSelectionModel(
-                DateRules.EveryDay(), TimeRules.Midnight,
-                dt => dt < EndDate.AddDays(-1) ? symbols : Enumerable.Empty<Symbol>()));
-
+            base.Initialize();
             SetAlpha(new EmaCrossAlphaModel());
-            SetPortfolioConstruction(new EqualWeightingPortfolioConstructionModel());
-            SetExecution(new ImmediateExecutionModel());
-            SetRiskManagement(new NullRiskManagementModel());
-        }
-
-        public override void OnEndOfAlgorithm()
-        {
-            // We have removed all securities from the universe. The Alpha Model should remove the consolidator
-            var consolidatorCount = SubscriptionManager.Subscriptions.Sum(s => s.Consolidators.Count);
-            if (consolidatorCount > 0)
-            {
-                throw new Exception($"The number of consolidator is should be zero. Actual: {consolidatorCount}");
-            }
         }
 
-        /// <summary>
-        /// This is used by the regression test system to indicate if the open source Lean repository has the required data to run this algorithm.
-        /// </summary>
-        public bool CanRunLocally { get; } = true;
-
-        /// <summary>
-        /// This is used by the regression test system to indicate which languages this algorithm is written in.
-        /// </summary>
-        public Language[] Languages { get; } = { Language.CSharp };
+        public override Language[] Languages { get; } = { Language.CSharp };
 
         /// <summary>
         /// Data Points count of all timeslices of algorithm
         /// </summary>
-        public long DataPoints => 14869;
+        public override long DataPoints => 14869;
 
         /// <summary>
         /// Data Points count of the algorithm history
         /// </summary>
-        public int AlgorithmHistoryDataPoints => 152;
+        public override int AlgorithmHistoryDataPoints => 152;
```

Comment:
I think `DataPoints/AlgorithmHistoryDataPoints/Languages` could be in the base class since they should be shared? I think Language should add python too?